packages/typespec-ts/src/modular/buildCodeModel.ts

```diff
@@ -841,10 +832,20 @@ function emitBasicOperation(
   } else {
     bodyParameter = emitBodyParameter(context, httpOperation);
     // Flatten the body parameter if it is an anonymous model
+    const originalBodyType = httpOperation.parameters.body.type;
+
     if (
       bodyParameter.type.type === "model" &&
-      bodyParameter.type.name === "" &&
-      bodyParameter.type.properties.length > 0
+      originalBodyType.kind === "Model" &&
+      originalBodyType.name === "" &&
+      [...originalBodyType.properties.keys()].every(
+        (k) =>
+          operation.parameters.properties.has(k) &&
+          (operation.parameters.properties.get(k) ===
+            (originalBodyType as Model).properties.get(k) ||
+            operation.parameters.properties.get(k) ===
+              (originalBodyType as Model).properties.get(k)?.sourceProperty)
```

Comment:
tbh i don't want to maintain this logic in modular side. this would introduce new debt for us when adopting to TCGC SdkPackage. i prefer to prioritize the task of integrating with SdkPackage for new changes in this semaster at least for our below goal.

```
we are committed to not accumulating any new debt moving forward.
```

@joheredi how do you think?

Reply:
https://github.com/Azure/autorest.typescript/issues/2660 issue tracked. I will have another pr to take this as an example.